homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -66,43 +68,66 @@ async def async_setup_platform(
-    httpsession = async_get_clientsession(hass)
-    train_api = TrafikverketTrain(httpsession, config[CONF_API_KEY])
-    sensors = []
-    station_cache = {}
+    """Import Trafikverket Train configuration from YAML."""
+    _LOGGER.warning(
+        # Config flow added in Home Assistant Core 2022.3, remove import flow in 2022.7
+        "Loading Trafikverket Train via platform setup is deprecated; Please remove it from your configuration"
+    )
+
     for train in config[CONF_TRAINS]:
-        try:
-            trainstops = [train[CONF_FROM], train[CONF_TO]]
-            for station in trainstops:
-                if station not in station_cache:
-                    station_cache[station] = await train_api.async_get_train_station(
-                        station
-                    )
-
-        except ValueError as station_error:
-            if "Invalid authentication" in station_error.args[0]:
-                _LOGGER.error("Unable to set up up component: %s", station_error)
-                return
-            _LOGGER.error(
-                "Problem when trying station %s to %s. Error: %s ",
-                train[CONF_FROM],
-                train[CONF_TO],
-                station_error,
+
+        new_config = {
+            CONF_API_KEY: config[CONF_API_KEY],
+            CONF_FROM: train[CONF_FROM],
+            CONF_TO: train[CONF_TO],
+            CONF_TIME: str(train.get(CONF_TIME)),
+            CONF_WEEKDAY: train.get(CONF_WEEKDAY, WEEKDAYS),
+        }
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=new_config,
             )
-            continue
-
-        sensor = TrainSensor(
-            train_api,
-            train[CONF_NAME],
-            station_cache[train[CONF_FROM]],
-            station_cache[train[CONF_TO]],
-            train[CONF_WEEKDAY],
-            train.get(CONF_TIME),
         )
-        sensors.append(sensor)
 
-    async_add_entities(sensors, update_before_add=True)
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Trafikverket sensor entry."""
+
+    httpsession = async_get_clientsession(hass)
+    train_api = TrafikverketTrain(httpsession, entry.data[CONF_API_KEY])
+
+    try:
+        to_station = await train_api.async_get_train_station(entry.data[CONF_TO])
+        from_station = await train_api.async_get_train_station(entry.data[CONF_FROM])
+    except ValueError as error:
+        if "Invalid authentication" in error.args[0]:
+            raise ConfigEntryAuthFailed from error
+        raise ConfigEntryNotReady(
+            f"Problem when trying station {entry.data[CONF_FROM]} to {entry.data[CONF_TO]}. Error: {error} "
+        ) from error
+
+    train_time = (
+        parse_time(entry.data.get(CONF_TIME, "")) if entry.data.get(CONF_TIME) else None
```

Comment:
We don't need a default in the first dict lookup if we have a condition that checks that the dict lookup returns something.